apps/client/src/pages/LocationPage.js

```diff
@@ -40,7 +42,21 @@ const LocationPage = ({ topic }) => {
         action: `postcode - ${slug.replace("-", " ")}`,
         name: address.postalCode.substring(0, 4),
       });
-      context.address = address;
+
+      // Load given answers from sessionContext
+      let answers = sessionContext.answers;
+
+      // Reset the checker and answers when the address is changed
+      if (answers && sessionAddress?.id !== address?.id) {
+        checkerContext.checker = null;
+        answers = null;
+      }
+
+      sessionContext.setSessionData({
+        address: { ...sessionContext.address, [slug]: address },
+        answers, // Either null or filled with given answers
+        questionIndex: 0, // Reset to 0 to start with the first question
+      });
```

Comment:
I think we should consider if this logic should be part of the LocationPage.

Reply:
Good point. I was thinking about this as well. But the PR is already going out of control. I will mark this as a separate ticket.